include/cjdb/contracts.hpp

```diff
@@ -4,10 +4,46 @@
 #ifndef CJDB_CONTRACTS_HPP
 #define CJDB_CONTRACTS_HPP
 
-#include <cstdio>
 #include <string_view>
 #include <type_traits>
 
+#ifdef _MSC_VER
+	#define CJDB_PRETTY_FUNCTION __FUNCSIG__
+	#define CJDB_FORCE_INLINE __forceinline
+#else
+	#define CJDB_PRETTY_FUNCTION __PRETTY_FUNCTION__
+	#define CJDB_FORCE_INLINE [[gnu::always_inline]] inline
+#endif // _MSC_VER
+
+#ifndef CJDB_PRINT_ERROR
+	#ifdef CJDB_USE_STDIO
+		#include <cstdio>
+
+		namespace cjdb::contracts_detail {
+			struct print_error_fn {
+				CJDB_FORCE_INLINE void operator()(std::string_view const message) const noexcept
+				{
+					std::fwrite(message.data(), sizeof(char), message.size(), stderr);
+				}
+			};
+			inline constexpr auto print_error = print_error_fn{};
+		} // namespace cjdb::contracts_detail
+	#else
+		#include <iostream>
+
+		namespace cjdb::contracts_detail {
+			struct print_error_fn {
+				CJDB_FORCE_INLINE void operator()(std::string_view const message) const noexcept
+				try {
+					std::cerr << message;
+				} catch(...) {}
```

Comment:
Thanks for making this change, but in light of the above, I'd prefer to have exactly one of formatted or unformatted output. The message path shouldn't be in a release build, so I don't really have a strong preference either way.

If you keep the formatted output, please delete the try/catch block: it's not really necessary for this context.

Reply:
The try-catch block is necessary to maintain the `noexcept`-ness of the existing API; writing to an `ostream` may throw an exception.